exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -41,7 +41,8 @@ type exporter struct {
 	retryCount       int
 	collectorID      string
 	svcStructuredLog *cwlogs.Client
-	pusher           cwlogs.Pusher
+	pusherOverride   cwlogs.Pusher
```

Comment:
Is this only used for testing? Is it necessary?  I'd think that the exporter can be initialized with a mock implementation of the `cwlogs.PusherCache` interface.

Reply:
I just wrote that and will be moving this to that